daemon/sync.go

```diff
@@ -0,0 +1,385 @@
+package daemon
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	"time"
+
+	"github.com/weaveworks/flux"
+	"github.com/weaveworks/flux/cluster"
+	"github.com/weaveworks/flux/event"
+	"github.com/weaveworks/flux/git"
+	"github.com/weaveworks/flux/resource"
+	fluxsync "github.com/weaveworks/flux/sync"
+	"github.com/weaveworks/flux/update"
+)
+
+type SyncTag interface {
```

Comment:
1. I don't think this needs to be exported -- it's defining the interface consumed here, rather than an interface to be mentioned elsewhere
 2. From the type name and the methods, I'd expect this to handle the sync tag (i.e., make the tag and push it) -- but that's done independently. Does it make any sense to put the sync tag update in the implementation too?
 3. I'm not sure `.Revision()` is used (but I have been wrong about this kind of thing before)

Reply:
All your comments made sense (and you were actually correct about the unused `.Revision()` this time :wink:).

I am not too happy about the amount of parameters `SetRevision` ended up with, but I see no other way at the moment, I may do a next gen refactor in a month or so.

Reply:
> I am not too happy about the amount of parameters `SetRevision` ended up with

I didn't think of that when making the suggestion -- I don't think it's too bad though. There will surely be some refactoring of this bit in the "read-only git" PR, so it'll get attention soon in any case.